lgn-worker/src/main.rs

```diff
@@ -199,25 +179,22 @@ async fn run_worker(
         Default::default()
     };
 
-    let mut provers_manager = tokio::task::block_in_place(move || -> Result<ProversManager> {
-        let mut provers_manager = ProversManager::new();
-        register_v1_provers(config, &mut provers_manager, &checksums)
-            .context("while registering provers")?;
-        Ok(provers_manager)
-    })
-    .context("creating prover managers")?;
```

Comment:
Move the registration to the manager constructor, this simplifies the API.

Reply:
Do we need the `block_in_place` now that everything in the worker is async? Can't we just `.await` the parameter setup?